examples/benchmark/profet_benchmark.py

```diff
@@ -0,0 +1,136 @@
+    name: str = "profet"
+    # Configuration options for the training of the meta-model used in the Profet tasks.
+    profet_train_config: MetaModelTrainingConfig = MetaModelTrainingConfig()
+
+    algorithms: List[str] = choice(*algos_available, default_factory=["random", "tpe"].copy)
+
+    # Number of repetitions for each experiment
+    n_repetitions: int = 10
+    # Optimization budget (max number of trials) for optimizing the task.
+    max_trials: int = 50
+    # Run in debug mode:
+    # - No presistent storage
+    # - More verbose logging
+    # (@TODO: This isn't technically correct: Benchmarks don't support the `debug` flag in
+    # `develop` but that feature is to be added by the long-standing warm-start PR.
+    debug: bool = False
+    # Random seed.
+    seed: int = 123
+    # Path to the pickledb file to use for the `storage` argument of the benchmark.
+    storage_pickle_path: Path = Path("profet.pkl")
+    # Path to the folder where figures are to be saved.
+    figures_dir: Path = Path("figures")
+
+    input_dir: Path = Path("profet_data")
+    checkpoint_dir: Path = Path("profet_outputs")
+
+    def __post_init__(self):
+        simple_parsing_logger = get_logger("simple_parsing")
+        if not self.debug:
+            simple_parsing_logger.setLevel(logging.WARNING)
+
+
+def main(config: ProfetExperimentConfig):
+    if issubclass(config.task_type, ProfetTask):
+        task = config.task_type(
+            max_trials=config.max_trials,
+            task_id=0,
+            train_config=config.profet_train_config,
+            input_dir=config.input_dir,
+            checkpoint_dir=config.checkpoint_dir,
+            seed=config.seed,
+        )
+    else:
+        # NOTE: This doesn't normally happen when using this from the command-line.
+        task = config.task_type(max_trials=config.max_trials)
+
+    print(f"Storage file used: {config.storage_pickle_path}")
+
+    benchmark = get_or_create_benchmark(
+        name=config.name,
+        algorithms=config.algorithms,
+        targets=[{"assess": [AverageResult(config.n_repetitions)], "task": [task,],}],
+        storage={
+            "type": "legacy",
+            "database": {"type": "pickleddb", "host": str(config.storage_pickle_path)},
+        },
+        debug=config.debug,
+    )
+    benchmark.setup_studies()
+
+    # Since we're using the QuadraticsTask, we can evaluate the 'similarity' between
+    # them, hence we just show these figures rather than the (very large) number of other
+    # potential figures.
+    figures_dir = config.figures_dir / benchmark.name
+    figures_dir.mkdir(exist_ok=True, parents=True)
+
+    benchmark.process(n_workers=1)
+
+    status = benchmark.status(False)
+    print(status)
+    # if all(isinstance(source_task, QuadraticsTask) for source_task in benchmark.source_tasks):
+    figures = benchmark.analysis()
+    # # # TODO: Instead of having like 30 figures, should try and create an interactive-ish
```

Comment:
Maybe would be worth making an issue? Otherwise this TODO may be forgotten. :)

Reply:
https://github.com/Epistimio/orion/issues/675